src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingsResponseMapper.java

```diff
@@ -72,13 +73,11 @@ private CaseHearing getCaseHearing(CaseHearingRequestEntity entity) {
         return caseHearing;
     }
 
-    private void setHearingResponseDetails(CaseHearing caseHearing, List<HearingResponseEntity> entities) {
-        for (HearingResponseEntity hearingResponseEntity : entities) {
-            caseHearing.setLastResponseReceivedDateTime(hearingResponseEntity.getRequestTimeStamp());
-            caseHearing.setResponseVersion(hearingResponseEntity.getHearingResponseId());
-            caseHearing.setHearingListingStatus(hearingResponseEntity.getListingStatus());
-            caseHearing.setListAssistCaseStatus(hearingResponseEntity.getListingCaseStatus());
-        }
```

Comment:
This was overwriting the same object in the loop, the whatever the last one is in the list (by chance) would determine the result being returned.